src/System.Private.CoreLib/src/System/RtType.cs

```diff
@@ -4857,17 +4848,20 @@ internal unsafe MdUtf8String(void* pUtf8String, int cUtf8String)
             m_StringHeapByteLength = cUtf8String;
         }
 
+        // Very common called version of the Equals pair
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         internal unsafe bool Equals(MdUtf8String s)
         {
+            bool isEqual = false;
```

Comment:
This implementation looks weird. Should it rather be something like this:
```csharp
{
    if (s.m_StringHeapByteLength != m_StringHeapByteLength)
         return false;
    return SpanHelpers.SequenceEqual<byte>(...);
}
```

(I doubt that the fast check for empty string is worth it. The empty strings should be pretty rare in reflection.)